test/e2e/alert_relabel_config_test.go

```diff
@@ -56,124 +61,70 @@ func TestAlertRelabelConfigTechPreview(t *testing.T) {
+		Separator:    ";",
+	}))
+
 	// Delete the AlertRelabelConfig.
-	err = relabelConfigs.Delete(ctx, arcName, metav1.DeleteOptions{})
+	err = relabelConfigs.Delete(ctx, arc.Name, metav1.DeleteOptions{})
 	if err != nil {
 		t.Fatal(errors.Wrap(err, "failed to delete AlertRelabelConfig"))
 	}
 
-	// Check that it is removed from the secret.
-	err = framework.Poll(time.Second, 2*time.Minute, func() error {
-		s, err := secrets.Get(ctx, relabelSecretName, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		var configsFromSecret []relabel.Config
-		err = yaml.Unmarshal([]byte(s.Data[relabelSecretKey]), &configsFromSecret)
-		if err != nil {
-			t.Fatal(errors.Wrap(err, "failed to unmarshal AlertRelabelConfig from secret"))
-		}
-
-		// Should only have the default config.
-		if len(configsFromSecret) != 1 {
-			return fmt.Errorf("Secret contains %d relabel configs, but only 1 expected",
-				len(configsFromSecret))
-		}
-
-		return nil
-	})
+	// Check Prometheus config forgot about the deleted AlertRelabelConfig.
+	validateCurrentRelabelConfig(t, initialRelabelConfig)
+}
 
+// prometheusRelabelConfig returns the alert relabel configuration part used by Prometheus config
+func prometheusRelabelConfig(t *testing.T) ([]*relabel.Config, error) {
+	prometheusConfig, err := f.PrometheusConfigFromSecret(t, f.Ns, prometheusConfigSecretName)
 	if err != nil {
-		t.Fatal(errors.Wrap(err, "failed to confirm relabel was removed from secret"))
+		return nil, err
 	}
+	return prometheusConfig.AlertingConfig.AlertRelabelConfigs, nil
 }
 
-func relabelConfigsEqual(arc osmv1.RelabelConfig, fromSecret relabel.Config) error {
-	if len(arc.SourceLabels) != len(fromSecret.SourceLabels) {
-		return fmt.Errorf("SourceLabels have different lengths (%d != %d)",
-			len(arc.SourceLabels), len(fromSecret.SourceLabels))
-	}
-
-	for i := range arc.SourceLabels {
-		if string(arc.SourceLabels[i]) != string(fromSecret.SourceLabels[i]) {
-			return fmt.Errorf("SourceLabel %d does not match", i)
+// validateCurrentRelabelConfig ensures that Prometheus config is using the expected relabel config
+func validateCurrentRelabelConfig(t *testing.T, expectedRelabelConfig []*relabel.Config) {
+	err := framework.Poll(time.Second, 1*time.Minute, func() error {
+		currentRelabelConfig, err := prometheusRelabelConfig(t)
+		if err != nil {
+			return err
 		}
-	}
-
-	if arc.Separator == "" {
-		arc.Separator = ";"
-	}
-
-	if arc.Separator != fromSecret.Separator {
-		return fmt.Errorf("Seperator does not match (%q != %q)", arc.Separator, fromSecret.Separator)
-	}
-
-	if arc.TargetLabel != fromSecret.TargetLabel {
-		return fmt.Errorf("TargetLabel does not match (%q != %q)", arc.TargetLabel, fromSecret.TargetLabel)
-	}
+		if !assert.ElementsMatch(t, expectedRelabelConfig, currentRelabelConfig) {
+			return errors.New("the expected relabel config is not applied yet.")
+		}
+		return nil
+	})
 
-	arcRegex, err := relabel.NewRegexp(arc.Regex)
 	if err != nil {
```

Comment:
(nit) could use `require.NoError(t, err)`?

Reply:
Now that there is no objection to testify, I migrated this and some others.